netebpfext/net_ebpf_ext_sock_ops.c

```diff
@@ -563,38 +560,34 @@ net_ebpf_extension_sock_ops_flow_delete(uint16_t layer_id, uint32_t callout_id,
         goto Exit;
     }
 
-    NET_EBPF_EXT_LOG_MESSAGE_UINT64(
-        NET_EBPF_EXT_TRACELOG_LEVEL_VERBOSE,
-        NET_EBPF_EXT_TRACELOG_KEYWORD_SOCK_OPS,
-        "Flow deleted.",
-        local_flow_context->parameters.flow_id);
-
     filter_context = local_flow_context->filter_context;
     if (filter_context == NULL) {
         goto Exit;
     }
 
-    if (local_flow_context->client_detached) {
-        // Since the hook client is detached, exit the function.
+    if (filter_context->base.client_detached) {
         goto Exit;
     }
 
-    attached_client = (net_ebpf_extension_hook_client_t*)filter_context->base.client_context;
-    if (attached_client == NULL) {
-        // This means that the eBPF program is detached and there is nothing to notify.
+    if (local_flow_context->client_detached) {
```

Comment:
Do we still need this specific check, now that we a similar flag in `filter_context.base` that we check above? If so, a comment explaining the reason would be of help.

Reply:
Not needed. Also  `net_ebpf_extension_sock_ops_wfp_flow_context_t` already contains `net_ebpf_extension_wfp_filter_context_t`. So please remove the duplicate `client_detached` field.